tests/test_optimizers.py

```diff
@@ -58,19 +67,19 @@ def test_different_sessions(self):
         with self.test_context() as session1:
             gpflow.reset_default_session()
             opt = self.optimizer()
-            opt.minimize(demo, maxiter=10)
+            opt.minimize(demo, maxiter=1)
```

Comment:
Maybe `maxiter=2`? One is often a bit of a special case... I've had situations before where my optimiser would run once but not twice. I don't feel strongly though.

Reply:
Here we are just checking that optimizer will not crash when model was created subject to it was initialized properly. But, later it would be good to add mocking (or introspection mechanism) which will check that session was called once (or more) depending on optimizer.